rosenpass/src/app_server.rs

```diff
@@ -1288,3 +1303,62 @@ impl crate::api::mio::MioManagerContext for MioManagerFocus<'_> {
         self.0
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn helper_support_ipv4_ipv6(sockets: &Vec<mio::net::UdpSocket>) -> bool {
+        let mut supports_iv4 = false;
+        let mut supports_ipv6 = false;
+        for socket in sockets {
+            if socket.local_addr().unwrap().is_ipv4() {
+                supports_iv4 = true;
+            } else {
+                supports_ipv6 = true;
+                if !socket.only_v6().unwrap() {
+                    supports_iv4 = true;
+                }
+            }
+        }
+        supports_iv4 && supports_ipv6
+    }
+
+    #[test]
+    pub fn test_open_sockets_empty() {
+        let mut sockets = Vec::new();
+        let _ = open_sockets(&mut sockets);
+        assert!(helper_support_ipv4_ipv6(&sockets))
+    }
+
+    #[test]
+    pub fn test_open_sockets_ipv4_localhost() {
+        let mut sockets = Vec::new();
+        sockets.push(
+            mio::net::UdpSocket::bind(SocketAddr::V4(SocketAddrV4::new(
+                Ipv4Addr::new(127, 0, 0, 1),
+                0,
+            )))
+            .unwrap(),
```

Comment:
```suggestion
        sockets.push(UdpSocket::bind("127.0.0.1:0".parse().unwrap()).unwrap());
```

Nit, IMHO this is more readable. Same for the IPv6 case.